src/main/java/at/hannibal2/skyhanni/data/PetAPI.kt

```diff
@@ -35,31 +186,429 @@ object PetAPI {
+        val oldPet = pet
+        pet = newPet
+        if (oldPet.arePetsEqual(newPet)) return
+        if (SkyHanniMod.feature.dev.debug.petEventMessages) {
+            ChatUtils.debug(oldPet.toString().replace("§", "&"))
+            ChatUtils.debug(newPet.toString().replace("§", "&"))
+        }
+        PetChangeEvent(oldPet, newPet).post()
+    }
+
+    @SubscribeEvent
+    fun onDebug(event: DebugDataCollectEvent) {
+        event.title("PetAPI")
+        if (pet != null) {
+            event.addIrrelevant {
+                add("petName: '${pet?.internalName}'")
+                add("petRarity: '${pet?.rarity}'")
+                add("petItem: '${pet?.petItem}'")
+                add("petHasSkin: '${pet?.hasSkin}'")
+                add("petLevel: '${pet?.level}'")
+                add("petXP: '${pet?.xp}'")
+                add("rawPetLine: '${pet?.rawPetName}'")
+            }
+        } else {
+            event.addData("no pet equipped")
+        }
+    }
+
+    @SubscribeEvent
+    fun onNEURepoReload(event: NeuRepositoryReloadEvent) {
+        val data = event.getConstant<NEUPetsJson>("pets")
+        xpLeveling = data.pet_levels
+        val xpLevelingCustomJson = data.custom_pet_leveling.getAsJsonObject()
+
+        xpLevelingCustom = xpLevelingCustomJson
+
+        petRarityOffset = data.pet_rarity_offset.getAsJsonObject().entrySet().associate { (rarity, offset) ->
+            (LorenzRarity.getByName(rarity) ?: LorenzRarity.ULTIMATE) to offset.asInt
+        }
+    }
+
+    private fun petNametoFakeInternalName(petName: String): String {
+        return when (petName.uppercase()) {
+            "T-REX" -> "TYRANNOSAURUS"
+            else -> petName.uppercase().replace(" ", "_")
+        }
+    }
+
+    private fun petNametoInternalName(petName: String, rarity: LorenzRarity): NEUInternalName {
+        return "${petNametoFakeInternalName(petName)};${rarity.id}".asInternalName()
+    }
+
+    fun PetData?.arePetsEqual(pet2: PetData?): Boolean {
+        return this?.internalName == pet2?.internalName &&
+            this?.cleanName == this?.cleanName &&
+            this?.rarity == this?.rarity &&
+            this?.petItem == this?.petItem &&
+            this?.hasSkin == this?.hasSkin &&
+            this?.level == this?.level &&
+            this?.rawPetName == this?.rawPetName
+    }
 }
+
+data class PetNBT(
+    @SerializedName("type") val type: String,
+    @SerializedName("active") val active: Boolean,
+    @SerializedName("exp") val exp: Double,
+    @SerializedName("tier") val tier: String,
+    @SerializedName("hideInfo") val hideInfo: Boolean,
+    @SerializedName("heldItem") val heldItem: String?,
+    @SerializedName("candyUsed") val candyUsed: Int,
+    @SerializedName("skin") val skin: String?,
+    @SerializedName("uuid") val uuid: String,
+    @SerializedName("uniqueId") val uniqueId: String,
+    @SerializedName("hideRightClick") val hideRightClick: Boolean,
+    @SerializedName("noMove") val noMove: Boolean
```

Comment:
serializedName is only needed when there is a difference between the serialized name and the variable name you want, that isnt the case here